src/main/java/de/qabel/desktop/hockeyapp/VersionClient.java

```diff
@@ -0,0 +1,138 @@
+
+        return parseVersionsResponse(responseContent);
+    }
+
+    HockeyAppVersion createVersion(String version) throws IOException, JSONException {
+        HttpPost request = requestBuilder.getHttpPost(API_VERSIONS_NEW);
+
+        List<NameValuePair> parameters = buildCreateParameters(version);
+        request.setEntity(new UrlEncodedFormEntity(parameters, HTTP.UTF_8));
+
+        HttpResponse response = requestBuilder.getHttpClient().execute(request);
+        if (response.getStatusLine().getStatusCode() != 201) {
+            throw new IOException("Create version(" + API_VERSIONS_NEW + ") failed! Wrong status code");
+        }
+        String responseContent = EntityUtils.toString(response.getEntity());
+
+        return parseVersionCreateResponse(responseContent);
+    }
+
+    List<HockeyAppVersion> parseVersionsResponse(String responseContent) throws IOException {
+        try {
+            versions = new LinkedList<>();
+            JSONObject parsedJson = new JSONObject(responseContent);
+            JSONArray versionArray = parsedJson.getJSONArray("app_versions");
+            for (int i = 0; i < versionArray.length(); i++) {
+                JSONObject jsonObj = versionArray.getJSONObject(i);
+
+//                String versionId = jsonObj.getString("id");
+                String shortVersion = jsonObj.getString("version");
+
+                versions.add(new HockeyAppVersion(shortVersion));
+            }
+            return versions;
+        } catch (JSONException e) {
+            throw new IOException("returned JSON was invalid", e);
+        }
+    }
+
+    HockeyAppVersion parseVersionCreateResponse(String responseContent) throws IOException {
+        try {
+            JSONObject parsedJson = new JSONObject(responseContent);
+            String versionId = parsedJson.getString("id");
+            String shortVersion = parsedJson.getString("version");
+
+            return new HockeyAppVersion(versionId, shortVersion);
+        } catch (JSONException e) {
+            throw new IOException("returned JSON was invalid", e);
+        }
+    }
+
+    List<NameValuePair> buildCreateParameters(String version) {
+        List<NameValuePair> parameters = new ArrayList<>();
+        parameters.add(new BasicNameValuePair("bundle_version", version));
+        parameters.add(new BasicNameValuePair("bundle_short_version", version));
+        return parameters;
+    }
+
+    HockeyAppVersion findVersion(String shortVersion) throws VersionNotFoundException, IOException {
+        if (shortVersion == null || shortVersion.isEmpty()) {
+            throw new VersionNotFoundException("Version: " + shortVersion + " not found!");
+        }
+        if (versions == null) {
+            versions = getVersions();
+        }
+        for (HockeyAppVersion version : versions) {
+            if (shortVersion.equals(version.getShortVersion())) {
+                return version;
+            }
+        }
+        return version;
```

Comment:
a loop over the versions with a return and _after_ that loop another return? Shouldn't there be a VersionNotFoundException instead?